client/src/components/ProfileSettings/MainPanel/MainPanel.tsx

```diff
@@ -4,21 +4,28 @@ import Payment from "../Payment/Payment";
 import ProfilePhoto from "../ProfilePhoto/ProfilePhoto";
 import Security from "../Security/Security";
 import Settings from "../Settings/Settings";
+import { useParams } from 'react-router-dom';
 
-const MainPanel = (panelIndex: any): JSX.Element => {
+const MainPanel = (): JSX.Element => {
+
+    const { componentId } = useParams<{ componentId: string }>();
 
     const menuItems = [
```

Comment:
It would be faster to use a switch/case statement here as opposed to filtering and then mapping through the array :)